src/client.rs

```diff
@@ -140,6 +140,7 @@ pub mod future {
               Resp: Deserialize + 'static,
               E: Deserialize + 'static,
     {
+        #[allow(unknown_lints, type_complexity)]
```

Comment:
Tells rustc to shut up about lints it doesn't know about, and tells clippy to shut up about its type complexity lint.